### ISSUE TITLE
remote control options

### ISSUE THREAD

**Issue by exerkaoss** [th-ch/youtube-music#704]
Thanks for the awesome work. I ve noticed remote control options are missing (ie touchportal connectivity).
will it be maybe added in a newer version?

![image](https://user-images.githubusercontent.com/75551439/164978069-46276f93-11a3-4c9c-a255-f2887e6df943.png)

Thanks again!

**Comment by Araxeus** [comment]
Where is this screenshot from?
my youtube music options don't look remotely close to that
![image](https://user-images.githubusercontent.com/78568641/164979561-46ad8d86-d6eb-4e34-8e9f-73387c38bf40.png)

**Comment by thijsi123** [comment]
Isn't that from this: https://github.com/ytmdesktop/ytmdesktop ?

**Comment by thijsi123** [comment]
Wrong place bud.

**Comment by exerkaoss** [comment]
> Wrong place bud.

is the touchportal plugin instructions on the official app version made by KillerBoss0219

**Comment by exerkaoss** [comment]
> Where is this screenshot from? my youtube music options don't look remotely close to that ![image](https://user-images.githubusercontent.com/78568641/164979561-46ad8d86-d6eb-4e34-8e9f-73387c38bf40.png)

the official one has this options since 1.13

**Comment by thijsi123** [comment]
> > Wrong place bud.
> 
> is the touchportal plugin instructions on the official app version made by KillerBoss0219

I'm sorry but I don't understand if you are asking or saying something to me.

--------------------------------------------------------------------------------------------------





> > Where is this screenshot from? my youtube music options don't look remotely close to that ![image](https://user-images.githubusercontent.com/78568641/164979561-46ad8d86-d6eb-4e34-8e9f-73387c38bf40.png)
> 
> the official one has this options since 1.13

You know that this is another project right?
If so, can you please explain what you are asking for? Like... is this a feature request or what?

**Comment by exerkaoss** [comment]
> > > Wrong place bud.
> > 
> > 
> > is the touchportal plugin instructions on the official app version made by KillerBoss0219
> 
> I'm sorry but I don't understand if you are asking or saying something to me.
> 
> > > Where is this screenshot from? my youtube music options don't look remotely close to that ![image](https://user-images.githubusercontent.com/78568641/164979561-46ad8d86-d6eb-4e34-8e9f-73387c38bf40.png)
> > 
> > 
> > the official one has this options since 1.13
> 
> You know that this is another project right? If so, can you please explain what you are asking for? Like... is this a feature request or what?

it is, i tried to ask in the first post "remote control will be added in a newer version?"

**Comment by thijsi123** [comment]
Mind sharing the link to your "first post"?

> > > > Wrong place bud.
> > > 
> > > 
> > > is the touchportal plugin instructions on the official app version made by KillerBoss0219
> > 
> > 
> > I'm sorry but I don't understand if you are asking or saying something to me.
> > > > Where is this screenshot from? my youtube music options don't look remotely close to that ![image](https://user-images.githubusercontent.com/78568641/164979561-46ad8d86-d6eb-4e34-8e9f-73387c38bf40.png)
> > > 
> > > 
> > > the official one has this options since 1.13
> > 
> > 
> > You know that this is another project right? If so, can you please explain what you are asking for? Like... is this a feature request or what?
> 
> it is, i tried to ask in the first post "remote control will be added in a newer version?" also, i m sorry i m kinda new to github so i m getting acquainted will all the functions and tabs..... :/

**Comment by exerkaoss** [comment]
> Mind sharing the link to your "first post"?
> 
> > > > > Wrong place bud.
> > > > 
> > > > 
> > > > is the touchportal plugin instructions on the official app version made by KillerBoss0219
> > > 
> > > 
> > > I'm sorry but I don't understand if you are asking or saying something to me.
> > > > > Where is this screenshot from? my youtube music options don't look remotely close to that ![image](https://user-images.githubusercontent.com/78568641/164979561-46ad8d86-d6eb-4e34-8e9f-73387c38bf40.png)
> > > > 
> > > > 
> > > > the official one has this options since 1.13
> > > 
> > > 
> > > You know that this is another project right? If so, can you please explain what you are asking for? Like... is this a feature request or what?
> > 
> > 
> > it is, i tried to ask in the first post "remote control will be added in a newer version?" also, i m sorry i m kinda new to github so i m getting acquainted will all the functions and tabs..... :/

it's my very first post 2h ago right here on top, with screenshots. i just thanked for the awesome work, pointed out the missing remote control option with screenshot to official version and asked if maybe it will be included in new releases.
Thanks again

**Comment by thijsi123** [comment]
Ahh, But this project isn't a fork of ' https://github.com/ytmdesktop/ytmdesktop '
This one is completely different and doesn't have features from ' https://github.com/ytmdesktop/ytmdesktop  '
You request a feature but it might not be implemented.

**Comment by Araxeus** [comment]
I personally probably won't implement this feature in the near future

maybe someone else will  ;)

**Comment by exerkaoss** [comment]
> Ahh, But this project isn't a fork of ' https://github.com/ytmdesktop/ytmdesktop ' This one is completely different and doesn't have features from ' https://github.com/ytmdesktop/ytmdesktop ' You may request a feature but it might not be implemented.

ah ok i see... but would be cool a remote control. i think it would complete the plugin list. then if not implemented, nevermind... i ll just hotkey it.

**Comment by exerkaoss** [comment]
> I personally probably won't implement this feature in the near future
> 
> maybe someone else will ;)

ah ok i see... but would be cool a remote control. i think it would complete the plugin list. then if not implemented, nevermind... i ll just hotkey it.

**Comment by thijsi123** [comment]
I don't even know what it does, and neither do I have a lot of experience in javascript (still learning).

**Comment by thijsi123** [comment]
Can you tell me?

**Comment by exerkaoss** [comment]
> Can you tell me?

the existing plugin for ytmd for touchportal hooks thru the app and lets you use all the basic yt functions + CC, volume, ffwd, rwd, add to playlist, fullscreen, switch between video/song version of what you are listening, as well as showing on the tablet or phone the actual video playing on ytmd + current/total song time.

**Comment by thijsi123** [comment]
Hhhm looks interesting, I'll take a look for myself.
I can't promise anything.

**Comment by Araxeus** [comment]
@thijsi123 holla at me if you want some help with javascript or with the structure of this app

can reach me on discord

**Comment by thijsi123** [comment]
Will do, my discord is 0AThijsL202#7411

**Comment by Araxeus** [comment]
and mine is on my GitHub profile ^^
> `Araxeus#0819`

**Comment by thijsi123** [comment]
Touch portal requires you to have a pro subscription to use plugins.
That's a no go.

**Comment by thijsi123** [comment]
You could just download touchportal and setup the free macros to use the volume and next song of the whole PC though.

**Comment by exerkaoss** [comment]
> You could just download touchportal and setup the free macros to use the volume and next song of the whole PC though.
that s what i did at the end. create a hotkey in the app from here and the a virtual keypress to those in touchportal, but at the same time lost all the above mentioned functions :D

**Comment by exerkaoss** [comment]
> and mine is on my GitHub profile ^^
> 
> > `Araxeus#0819`

> Touch portal requires you to have a pro subscription to use plugins. That's a no go.

i have it  pro but so due to this and all we talked already, would it be possible to add more hotkeys to the 3 existing ones?

**Comment by thijsi123** [comment]
> > and mine is on my GitHub profile ^^
> > > `Araxeus#0819`
> 
> > Touch portal requires you to have a pro subscription to use plugins. That's a no go.
> 
> i have it pro but so due to this and all we talked already, would it be possible to add more hotkeys to the 3 existing ones?
> 
> like for example download current song volume shuffle repeat

No sorry, I can't import the plugin that means that I can't test the code at all.

**Comment by Araxeus** [comment]
You could set custom hotkeys in the `shortcuts` plugin

then use those hotkeys from touch portal

download current song isn't available there but all other things you said are

https://github.com/th-ch/youtube-music/blob/23058729f3965deb0bff9b638feb5d419da6e913/providers/song-controls.js#L12-L48

**Comment by thijsi123** [comment]
Thanks, I'll try that.

**Comment by exerkaoss** [comment]
> > i have it pro but so due to this and all we talked already, would it be possible to add more hotkeys to the 3 existing ones?
> > like for example
> > download current song
> > volume
> > shuffle
> > repeat
> 
> @exerkaoss you could set custom hotkeys in the `shortcuts` plugin
> 
> then use those hotkeys from touch portal
> 
> download current song isn't available there but all other things you said are
> 
> https://github.com/th-ch/youtube-music/blob/23058729f3965deb0bff9b638feb5d419da6e913/providers/song-controls.js#L12-L48

i cannot thank you enough! <3
i was already solving volume control assigning a different channel to the app in win and hooking it on voicemeeter potato but your answers solves a LOT!!!!

**Comment by Araxeus** [comment]
> i cannot thank you enough! <3 i was already solving volume control assigning a different channel to the app in win and hooking it on voicemeeter potato but your answers solves a LOT!!!!

if you use the `precise-volume` plugin then you can even set shortcuts for your specific 'steps'
(see in the plugin options)

**Comment by exerkaoss** [comment]
> hotkeys in the `shortcuts` plugin

i m sorry but is it normal i see just this 3 hotkeys?
![image](https://user-images.githubusercontent.com/75551439/164986786-842e4186-12b2-41d3-b0b0-d908f1f0d50a.png)

**Comment by Araxeus** [comment]
yes like I said, if you want to use the more advanced song controls, you have to do it via config.json
like so:
```json
		"shortcuts": {
			"enabled": true,
			"overrideMediaKeys": false,
			"global": {
				"previous": "Ctrl+Left",
				"playPause": "Ctrl+Space",
				"next": "Ctrl+Right",
				"like": "Ctrl+L",
				"go10sForward": "Ctrl+Shift+Right",
			}
		},
```

do it when the app is off or it might overwrite the settings you write

and you can easily access the config.json file via
![image](https://user-images.githubusercontent.com/78568641/164986992-7e83bdd6-c297-4846-8d1e-ddecd580d471.png)

and again like I said you can easily set shortcuts for volume in `precise-volume` if you want
![image](https://user-images.githubusercontent.com/78568641/164987040-1e6ef050-7e0c-4554-906a-a684a8289876.png)

**Comment by Araxeus** [comment]
oh and you can see valid values for your shortcut keys in
https://www.electronjs.org/docs/latest/api/accelerator#available-modifiers

**Comment by exerkaoss** [comment]
> yes like I said, if you want to use the more advanced song controls, you have to do it via config.json like so:
> 
> ```json
> 		"shortcuts": {
> 			"enabled": true,
> 			"overrideMediaKeys": false,
> 			"global": {
> 				"previous": "Ctrl+Left",
> 				"playPause": "Ctrl+Space",
> 				"next": "Ctrl+Right",
> 				"like": "Ctrl+L",
> 				"go10sForward": "Ctrl+Shift+Right",
> 			}
> 		},
> ```
> 
> do it when the app is off or it might overwrite the settings you write
> 
> and you can easily access the config.json file via ![image](https://user-images.githubusercontent.com/78568641/164986992-7e83bdd6-c297-4846-8d1e-ddecd580d471.png)
> 
> and again like I said you can easily set shortcuts for volume in `precise-volume` if you want ![image](https://user-images.githubusercontent.com/78568641/164987040-1e6ef050-7e0c-4554-906a-a684a8289876.png)

again thank you so much!

**Comment by thijsi123** [comment]
Oh it was that easy😅
I'm glad that you got it to work.

**Comment by Araxeus** [comment]
> again thank you so much!

You're welcome. 😃

Feel free to ask if you still need help

**Comment by exerkaoss** [comment]
> > again thank you so much!
> 
> You're welcome. 😃
> 
> Feel free to ask if you still need help

unfortunately i do.
![image](https://user-images.githubusercontent.com/75551439/164987981-598be758-2622-4c6f-9acc-2468f88b177d.png)

the added lines generate this error, if i cancel them the app goes again without problems :/
did i write some syntax wrong? i don t think...

**Comment by Araxeus** [comment]
> did i write some syntax wrong? i don t think...

yes, JSON object can't end with <kbd>,</lbd>

remove the <kbd>,</lbd>after `switchRepeat`

**Comment by exerkaoss** [comment]
> > did i write some syntax wrong? i don t think...
> 
> yes, JSON object can't end with ,
> 
> remove the last , (after `switchRepeat`)
> 
> if you used a normal code editor (like [vscode](https://code.visualstudio.com/)) it would highlight the issue automatically 😋

![278820242_691345558843012_4230942143968981763_n](https://user-images.githubusercontent.com/75551439/165079309-1835bbd3-ff06-42c2-8c1c-5e1bc48d8d02.jpg)

took me a while and some "ifs"+the json config but managed to let everything work. also the 500ms text updater through another plugin that i use to show what i'm listening to my stream :D
Thanks again for the support shown yesterday!

**Comment by Zo-Bro-23** [comment]
Will be fixed by #1094.

**Comment by JellyBrick** [comment]
Duplicate of #739
